#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "sensor"


gen = ParameterGenerator()

gen.add("switch_imu1_bool", bool_t, 0, "switch of IMU 1", True)
gen.add("switch_imu2_bool", bool_t, 0, "switch of IMU 2", True)
gen.add("switch_imu3_bool", bool_t, 0, "switch of IMU 3", False)
gen.add("maxsigma_head", double_t, 0, "superier of variance", 1000, 0, 10000)
gen.add("minsigma_head", double_t, 0, "inferier of variance", 0, 0, 10)
gen.add("maxsigma_roll", double_t, 0, "superier of variance", 1000, 0, 10000)
gen.add("minsigma_roll", double_t, 0, "inferier of variance", 0, 0, 10)
gen.add("maxsigma_pitch", double_t, 0, "superier of variance", 10000, 0, 10000)
gen.add("minsigma_pitch", double_t, 0, "inferier of variance", 0, 0, 10)


log_enum = gen.enum([gen.const("info", int_t, 0, "log print flag:INFO"),
                     gen.const("warn", int_t, 1, "log print flag:WARN"),
                     gen.const("error", int_t, 2, "log print flag:ERROR")],
                    "Set Log Level")
gen.add("log_level", int_t, 0, "Set Log Level", 0, 0, 2, edit_method=log_enum)

exit(gen.generate(PACKAGE, "imu_fusion", "imu_fusion"))
